client/clientservice/src/request_service.cpp

```diff
@@ -22,6 +22,48 @@ namespace concord::client::clientservice {
 
 namespace requestservice {
 
+struct GrpcMetadataCarrierForReading : opentracing::TextMapReader {
```

Comment:
This class doesn't seem to be specific to the request service. Can it be used by other components? If so then we should put this code into a common/utility space. Can you maybe re-use or refactor the code in `client/thin-replica-client/src/trace_contexts.cpp`?

Reply:
Refactored the code to make it reusable as suggested by the above comment.